my-app/src/main/java/com/mycompany/app/App.java

```diff
@@ -1,15 +1,36 @@
 package com.mycompany.app;
 
 import io.usethesource.vallang.IInteger;
+import io.usethesource.vallang.IString;
 import io.usethesource.vallang.IValueFactory;
 
+import org.rascalmpl.exceptions.RuntimeExceptionFactory;
+
+import frink.errors.FrinkEvaluationException;
+import frink.parser.Frink;
 
 public class App 
 {
     private final IValueFactory vf;
+    private Frink interp;
 
     public App(IValueFactory vf) {
         this.vf = vf;
+        this.interp = new Frink();
+
+        // Enable restrictive mode, since we cannot trust user input
+        // Enabling restrictive mode however prevents Frink itself from accessing its internal units.txt file :(
+        // However, we can first load these units by doing a dummy calculation
+        try {
+			this.interp.parseString("1/2 c");
+		} catch (FrinkEvaluationException e) {
+			System.err.print("Frink could not convert the dummy input: 1/2 c");
+			e.printStackTrace();
+		}
+
+        // Now that the units are loaded, we can enable restrictive mode
+        this.interp.setRestrictiveSecurity(true);
+
```

Comment:
This is Frink-specific. Note that this is still the location to initialize classes when using other libraries.